examples/t9/t9_streaming.cc

```diff
@@ -0,0 +1,431 @@
+    std::lock_guard<std::mutex> obolus(charon);  // <<<<<<<<<< mutex
+    const Node& node = baseT::get<0>(t);
+    const double boxsize = 2.0 * L * pow2(-key.n);
+    sumsq += (node.s * node.s + node.d * node.d) * boxsize;
+  }
+
+  double get() const {
+    double value = sumsq;
+    ttg_sum(ttg_default_execution_context(), value);
+    return std::sqrt(value);
+  }
+};
+
+auto make_norm2(const nodeEdge& in) { return std::make_unique<Norm2>(in); }  // for dull uniformity
+
+auto make_start(const ctlEdge& ctl) {
+  auto func = [](const Key& key, std::tuple<ctlOut>& out) { send<0>(key, Control(), out); };
+  return wrap<Key>(func, edges(), edges(ctl), "start", {}, {"control"});
+}
+
+// Operations used with BinaryOp
+double add(double a, double b) { return a + b; }
+
+double sub(double a, double b) { return a - b; }
+
+double mul(double a, double b) { return a * b; }
+
+// Functions we are testing with
+double A(const double x) { return std::exp(-x * x); }
+
+double diffA(const double x) { return -2.0 * x * exp(-x * x); }
+
+double B(const double x) { return std::exp(-x * x) * std::cos(x); }
+
+double C(const double x) { return std::exp(-x * x) * std::sin(x); }
+
+double R(const double x) { return (A(x) + B(x)) * C(x); }
+
+int main(int argc, char** argv) {
+  ttg_initialize(argc, argv, -1);
+  {
+    ctlEdge ctl("start ctl");
+    nodeEdge a("a"), b("b"), c("c"), abc("abc"), diffa("diffa"), errdiff("errdiff"), errabc("errabc"), a_plus_b("a+b"),
+        a_plus_b_times_c("(a+b)*c"), deriva("deriva"), compa("compa"), recona("recona");
+
+    // The following can indeed be specified in any order!
+    auto p1 = make_project(&A, ctl, a, "project A");
+    auto p2 = make_project(&B, ctl, b, "project B");
+    auto p3 = make_project(&C, ctl, c, "project C");
+    auto p4 = make_project(&R, ctl, abc, "project ABC");
+    auto p5 = make_project(&diffA, ctl, diffa, "project dA/dx");
+
+    auto b1 = make_binary_op(add, a, b, a_plus_b, "a+b");
+    auto b2 = make_binary_op(mul, a_plus_b, c, a_plus_b_times_c, "(a+b)*c");
+    auto b3 = make_binary_op(sub, a_plus_b_times_c, abc, errabc, "(a+b)*c - abc");
+    auto b4 = make_binary_op(sub, diffa, deriva, errdiff, "dA/dx analytic - numeric");
+
+    auto d = make_diff(a, deriva, "dA/dx numeric");
+
+    auto norma = make_norm2(a);
+    auto normabcerr = make_norm2(errabc);
+    auto normdifferr = make_norm2(errdiff);
+
+    auto comp1 = make_compress(a, compa, "compress(A)");
+    auto &reduce_leaves_op = std::get<1>(comp1);
+
+    reduce_leaves_op->set_input_reducer<0>([](Node &&left, Node &&right) {
+                                           double s = (left.s + right.s) * 0.5;
+                                           double d = (left.s - right.s) * 0.5;
+                                           return Node(left.key, s, d, true);
+                                         });
+    //Key really doesn't matter here, but for compilation, we need to use the overload. FIX THIS!!
```

Comment:
@pnookala not following the issue ... is the comment still relevant?

Reply:
The comment is not valid anymore. I will clean it up.